curriculum/challenges/english/blocks/workshop-shortest-path-algorithm/687b3ef0c2fe185b2abc4654.md

```diff
@@ -0,0 +1,113 @@
+---
+id: 687b3ef0c2fe185b2abc4654
+title: Step 20
+challengeType: 20
+dashedName: step-20
+---
+
+# --description--
+
+Once the algorithm has finished running, you need to decide which node(s) to display results for.
+
+If a specific `target_node` was provided, you'll only show the distance and path to that node. Otherwise, you'll show results for all nodes.
+
+After the outer `for _ in range(n):` loop ends, create a variable named `targets`. Use a conditional expression to assign it `[target_node]` if `target_node` is not `None`, otherwise assign it `list(range(n))`.
+
+# --hints--
+
+You should create a variable named `targets`.
+
+```js
+({
+    test: () => {
+      assert(
+        runPython(
+          `_Node(_code).find_function("shortest_path").has_variable("targets")`
+        )
+      );
+    }
+});
+```
+
+Your `targets` variable should use a conditional expression with `if` and `else`.
+
+```js
+({
+  test: () => {
+      runPython(
+        `"if" in str(_Node(_code).find_function("shortest_path").find_variable("targets")) and "else" in str(_Node(_code).find_function("shortest_path").find_variable("targets"))`
+    );
+  }
+});
+```
+
+The conditional expression should check if `target_node is not None`.
+
+```js
+({
+  test: () => {
+      runPython(
+        `"target_node is not None" in str(_Node(_code).find_function("shortest_path").find_variable("targets"))`
+    );
+  }
+});
+```
```

Comment:
````suggestion
Your `targets` variable should contain a conditional expression.

```js
({
    test: () => {
      runPython(`
        import ast
        val =_Node(_code).find_function("shortest_path").find_variable("targets").tree.value
        assert isinstance(val, ast.IfExp)
      `);
    }
});
```

The conditional expression should check if `target_node is not None`.

```js
({
    test: () => {
        runPython(`
        compare = _Node(_Node(_code).find_function("shortest_path").find_variable("targets").tree.value.test)
        assert compare.is_equivalent("target_node is not None")
        `);
    }
});
```
````